### ISSUE TITLE
High-contrast mode: buttons with `.disabled` class look active in high-contrast mode

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#7449]
This is a bug specific to [Windows high-contrast mode](https://blogs.windows.com/msedgedev/2020/09/17/styling-for-windows-high-contrast-with-new-standards-for-forced-colors/). cc @kbayliss who did the testing for this.

Button with `.disabled` class, rather than attribute:

![7449](https://user-images.githubusercontent.com/877585/129969904-fde7b6eb-7638-4422-84fc-913c35bf30e8.png)

We should instead use the `disabled` attribute on a `<button>` element, or if not possible add high-contrast mode styles to apply the correct "disabled" system colors.

Additionally the icon would need to match the colour of the text.

**Comment by krisinvorce** [comment]
Hi, I would love to work on this issue, could you assign it to me?

**Comment by mr-vaibh** [comment]
#7794

**Comment by mr-vaibh** [comment]
#7794 tried to FIX

**Comment by TUSHAR-VERMA-star** [comment]
Can I take this issue ?

**Comment by PaarthAgarwal** [comment]
I'd like to work on this issue

**Comment by zerolab** [comment]
If there is no related PR, and the issue doesn't have the "Design decision needed" label, then just do it. No need to ask

**Comment by thibaudcolas** [comment]
Assigning to @anujaraj as part of her project.

Looking into this further, I believe the simplest fix would be to add the following styles to our `disabled` styles, in forced colors mode: https://github.com/wagtail/wagtail/blob/4a7892a9b53c0f17f1f9eec6d579d8cea75d7941/client/scss/components/_button.scss#L275-L282

- `color: GrayText;`
- `border-color: curentColor;`

This needs a bit more testing to check this achieves the desired effect though.
